### ISSUE TITLE
create_tables with safe=True doesn't work for indexes

### ISSUE THREAD

**Issue by miguelglopes** [coleifer/peewee#2079]
When running the following code:

```
import peewee

db = peewee.MySQLDatabase(
    config["database"]["db"],
    user=config["database"]["user"],
    password=config["database"]["pwd"],
    host=config["database"]["host"],
    port=config["database"]["port"],
)

db.connect()

class Show(peewee.Model):
    showid = peewee.PrimaryKeyField()
    year = peewee.IntegerField(null=True)

    class Meta:
        db_table = "Show"
        database = db

class LatestEpisode(peewee.Model):
    show = peewee.ForeignKeyField(Show, backref="latestEpisodes")
    season = peewee.IntegerField()
    episode = peewee.IntegerField()

    class Meta:
        db_table = "LatestEpisode"
        database = db

db.create_tables([Show, LatestEpisode], safe=True)
```

Everything runs as expected if the tables don't exist.

However, if they exist, i got the following error:
peewee.InternalError: (1061, "Duplicate key name 'latestepisode_show_id'")

The problem is that, although the tables are created with "IF NOT EXISTS", due to the safe=True, the indexes are not:
('CREATE INDEX `latestepisode_show_id` ON `LatestEpisode` (`show_id`)', [])

**Comment by coleifer** [comment]
MySQL does not support create index if not exists. Since you're using MySQL, just modify the code to check for the tables' existence first.

**Comment by coleifer** [comment]
MySQL does not support "IF NOT EXISTS" for indexes for some reason.

So what peewee does when you're using MySQL is to first check if the table exists -- and if it does -- then it's a no-op. So I think something weird may be going on in your setup.

**Comment by miguelglopes** [comment]
Ok, I think I found the problem.

I capitalized the first letter in the name of the table. But on mysql workbench it showed as all lower letters. I did some tests:
table._meta.table_name returned 'Show'
db.table_exists("Show") returned false 
db.table_exists("show") returned true

If I capitalize the first letter of the name of the table, it doesn't detect the table exists and does the create table operation.

If i don't capitalize, everything works as expected.
